src/common/cli_wrapper.h

```diff
@@ -0,0 +1,275 @@
+#pragma once
+
+#include "3rd_party/CLI/CLI.hpp"
+#include "3rd_party/yaml-cpp/yaml.h"
+#include "common/any_type.h"
+#include "common/definitions.h"
+#include "common/logging.h"
+
+#include <iostream>
+#include <map>
+#include <string>
+
+namespace marian {
+namespace cli {
+
+// try to determine the width of the terminal
+static uint16_t guess_terminal_width(uint16_t max_width = 0,
+                                     uint16_t default_width = 180);
+
+namespace validators {
+const CLI::detail::ExistingFileValidator file_exists;
+const CLI::detail::ExistingDirectoryValidator dir_exists;
+const CLI::detail::ExistingPathValidator path_exists;
+
+const CLI::detail::NonexistentPathValidator path_not_exists;
+
+typedef CLI::Range range;
+}
+
+class CLIFormatter : public CLI::Formatter {
+public:
+  CLIFormatter(size_t columnWidth, size_t screenWidth);
+  virtual std::string make_option_desc(const CLI::Option *) const;
+
+private:
+  size_t screenWidth_{0};
+};
+
+class CLIWrapper {
+private:
+  // Stores option variables
```

Comment:
I disagree with line-end comments as these decrease readability even if formatted for 80 or 100 character-length lines. I've updated comments

Reply:
I disagree with vertical comments, as they decrease readability since less code fits on a screen.

The thing to do is to horizontally align the comments. Some auto-formatters, like the one we used in CNTK, can do that. Then they are very nicely readable, while getting out of the way of the actual source code. This is the standard we should follow.